### ISSUE TITLE
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'

### ISSUE THREAD

**Issue by statiksof** [NCAS-CMS/cf-training#1]
I wanted to try this [notebook](https://github.com/NCAS-CMS/cf-training/blob/master/full_course/notebooks/contour_plots.ipynb) but when executing this:

```
# Make a contour plot of the zonal mean of this data
cfp.con(f.collapse('mean','longitude'))
```
I got this error:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-3-0cdd342646ad> in <module>
      1 # Make a contour plot of the zonal mean of this data
----> 2 cfp.con(f.collapse('mean','longitude'))

/python3.7/site-packages/cfplot/cfplot.py in con(f, x, y, fill, lines, line_labels, title, colorbar_title, colorbar, colorbar_label_skip, ptype, negative_linestyle, blockfill, zero_thick, colorbar_shrink, colorbar_orientation, colorbar_position, xlog, ylog, axes, xaxis, yaxis, xticks, xticklabels, yticks, yticklabels, xlabel, ylabel, colors, swap_axes, verbose, linewidths, alpha, colorbar_text_up_down, colorbar_text_down_up, colorbar_drawedges, colorbar_fraction, colorbar_thick, colorbar_anchor, colorbar_labels, linestyles, zorder)
   1283                     fmt=fmt,
   1284                     colors=colors,
-> 1285                     fontsize=text_fontsize)
   1286 
   1287                 # Thick zero contour line

/python3.7/site-packages/matplotlib/axes/_axes.py in clabel(self, CS, levels, **kwargs)
   6361             All other parameters are documented in `~.ContourLabeler.clabel`.
   6362         """
-> 6363         return CS.clabel(levels, **kwargs)
   6364 
   6365     #### Data analysis

/python3.7/site-packages/matplotlib/contour.py in clabel(self, levels, fontsize, inline, inline_spacing, fmt, colors, use_clabeltext, manual, rightside_up, zorder)
    149         self.rightside_up = rightside_up
    150         if zorder is None:
--> 151             self._clabel_zorder = 2+self._contour_zorder
    152         else:
    153             self._clabel_zorder = zorder

TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'
```
I tried with:

`cf-python` 3.10 and 3.6.0
`cf-plot` 3.0.25 and 3.0.20
`matplotlib` 3.3.2
`udunits2` 2.2.27.6 and 2.2.26

**Comment by ajheaps** [comment]
I've just uploaded cf-plot 3.0.29 to GitHub, pypi and the ncas conda channel and this should fix this issue.  Please let me know if it doesn't.  

Regards
Andy

**Comment by statiksof** [comment]
@ajheaps unfortunately, I have the same error with `3.0.29`.

**Comment by ajheaps** [comment]
Thanks for getting back to me.  I can now replicate this issue using matplotlib 3.3.2.  Try downgrading to 3.1.3 as that works for me.

I'll look into providing a patch later on this week if you find it looks okay at matplotlib 3.1.3.

Regards
Andy

**Comment by statiksof** [comment]
@ajheaps I confirm, downgrading to `matplotlib 3.1.3` solves the issue. It will be great if the patch works with `matplotlib 3.3.2`.

**Comment by ajheaps** [comment]
I have fixed this issue in cf-plot 3.0.32.  The problem was internally the plotting order of the plots was set to None and this caused  an error in the newer versions of matplotlib.  I have now set the zorder of plots to default to 1.  cf-plot 3.0.32 is available on GitHub, pypi and the ncas conda channel.

**Comment by sadielbartholomew** [comment]
That's excellent, thanks Andy. I'll close this now since, unless someone reports otherwise by comment here or elsewhere, the issue in question has been resolved.
